### PR TITLE
Resources stockpiling

### PR DESCRIPTION
**Pull request by ipanic**
- Ability to stockpile food and raw resources to the warehouse, not granary/workshop
- Ability to disable warehouse cart pusher to halt movement stockpiled resources
Updated version of this PR: https://github.com/Keriew/augustus/pull/293

### CODE REVIEW

**Review by Keriew — CHANGES_REQUESTED** [review]
Looks good codewise aside from a few small issues, could you update it to the latest build so I can merge it.

**Comment by ipanic** [comment]
> Looks good codewise aside from a few small issues, could you update it to the latest build so I can merge it.

Well, I'll fix issues next week

File: src/building/building.c [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "core/array.h"
 #include "core/log.h"
 #include "figure/formation_legion.h"
+#include "industry.h"
```
Comment (Keriew) [inline src/building/building.c:17]: Please add the path to it (building/industry.h) and place it in the alphabetical order of the includes.

File: src/translation/translation.h [hunk 2]
```diff
@@ -697,7 +697,10 @@ typedef enum {
     TR_WINDOW_BUILDING_DISTRIBUTION_CART_PUSHER_GETTING,
     TR_BUILDING_PALISADE,
     TR_BUILDING_PALISADE_DESC,
-    TRANSLATION_MAX_KEY
+    TRANSLATION_MAX_KEY,
```
Comment (Keriew) [inline src/translation/translation.h:700]: TRANSLATION_MAX_KEY has to be the last one.

File: src/window/building/distribution.c [hunk 3]
```diff
@@ -1076,6 +1122,15 @@ static void toggle_partial_resource_state(int index, int param2)
     window_invalidate();
 }
 
+static void button_stockpiling(int param1, int param2)
+{
+    building* b = building_get(data.building_id);
```
Comment (Keriew) [inline src/window/building/distribution.c:1127]: Please attach pointer to the variable name, not to the type.

**Review by Keriew — CHANGES_REQUESTED** [review]

**Comment by Keriew** [comment]
Looks good aside from the minor formatting issues!

**Comment by Keriew** [comment]
Thanks, it's good now! 

There seems to be a conflict preventing me from merging, could you resolve it?

**Comment by ipanic** [comment]
> Thanks, it's good now!
> 
> There seems to be a conflict preventing me from merging, could you resolve it?

Done

**Review by Keriew — APPROVED** [review]

**Comment by Keriew** [comment]
Thanks, this will be useful!

**Pull request merged**